lib/ui/painting/canvas.cc

```diff
@@ -105,6 +105,7 @@ void Canvas::saveLayerWithoutBounds(const Paint& paint,
   if (!canvas_) {
     return;
   }
+  TRACE_EVENT0("flutter", "Canvas::saveLayer");
```

Comment:
This is incorrect. This line is only recording a saveLayer and has nothing to do with the execution of saveLayers.

The recorded picture can be reused on every single frame from here on and it won't be recorded (there's not much that can be done for that with SkP, but DL could record them as they execute).

The recorded picture could be a mistake and never used even once.

This trace will appear on the build thread, not the raster thread.

There is everything wrong with this event.

Reply:
I know that this record won't be enough, but in most cases it will provide signal about where in developer code someone actually triggered a saveLayer. My more recent concern is that it might not be enough, since things like `drawColor` and `drawPaint` might also trigger a saveLayer. 

How would you suggest achieving this?